totp.go

```diff
@@ -199,6 +205,69 @@ func (a *App) DeleteTOTP(w http.ResponseWriter, r *http.Request) {
 	jsonResponse(w, nil, http.StatusNoContent)
 }
 
+// ValidateTOTP is the http handler to validate a passcode.
+func (a *App) ValidateTOTP(w http.ResponseWriter, r *http.Request) {
+	const notFound = "TOTP not found"
+
+	requestBody, err := parseValidateTOTPRequestBody(r.Body)
+	if err != nil {
+		log.Printf("Invalid ValidateTOTP request body: %s", err)
+		jsonResponse(w, "Invalid request", http.StatusBadRequest)
+		return
+	}
+
+	id := r.PathValue(UUIDParam)
+
+	key, err := getAPIKey(r)
+	if err != nil {
+		jsonResponse(w, fmt.Errorf("API Key not found in request context: %w", err), http.StatusInternalServerError)
+		return
+	}
+
+	var t TOTP
+	err = a.db.Load(envConfig.TotpTable, TOTPTablePK, id, &t)
+	if err != nil {
+		if strings.Contains(err.Error(), "does not exist") {
+			jsonResponse(w, notFound, http.StatusNotFound)
+		} else {
+			log.Printf("error loading TOTP: %s", err)
+			jsonResponse(w, "Internal server error", http.StatusInternalServerError)
+		}
+		return
+	}
+
+	if key.Key != t.ApiKey {
+		jsonResponse(w, notFound, http.StatusNotFound)
+		return
+	}
+
+	valid := totp.Validate(requestBody.Code, t.Key)
+	if !valid {
+		jsonResponse(w, "Invalid", http.StatusUnauthorized)
+		return
+	}
+
+	jsonResponse(w, "Valid", http.StatusOK)
+}
+
+// parseValidateTOTPRequestBody parses and validates the ValidateTOTP request body
+func parseValidateTOTPRequestBody(body io.ReadCloser) (requestBody *ValidateTOTPRequestBody, err error) {
+	if body == nil {
+		return nil, fmt.Errorf("empty request body")
+	}
+
+	err = json.NewDecoder(body).Decode(&requestBody)
```

Comment:
`requestBody` already a pointer

```suggestion
	err = json.NewDecoder(body).Decode(requestBody)
```